pippy/PipelineStage.py

```diff
@@ -199,29 +206,27 @@ def __init__(
         # Caching chunk outputs for final output merge or reduction
         self.output_chunks: List[Any] = []
 
-        # Find my submodule
-        self.split_gm = self.pipe.split_gm
-        named_children = list(self.split_gm.named_children())
-        self.name, self.submod = named_children[stage_index]
+        # Find stage nodes in graph
+        submod_nodes = [
+            node for node in pipe_info.graph.nodes if node.op == "call_module"
```

Comment:
i think this PR is good on its own as a refactoring step that introduces a good API boundary, which is good.

but i'm curious, for stuff like this, where you still reach into pipe_info.graph, do you plan to eventually reduce this over time and replace with 'well defined fields' on pipe_info?

Reply:
what would be a better way to describe stage relationship than a graph?

Reply:
note that this is an unflattened graph, so all it has is just:
submod0 --> submod1 --> submod2 ...

Reply:
I could stick `pipe_info` into `stage_module`. But my concern is if any of DDP, FSDP, TP wraps `stage_module`, this attribute can't be found by `_PipelineStage`. Any good solution along this direction?

Reply:
i guess turning it from a graph (which is general) into specific fields like "my receivers" and "my senders" - which is specific and verbose but also more well constrained

i'm not too familiar with all the things you use the graph for though, so i dont know how much this will work